ExternalModule.php

```diff
@@ -163,27 +208,51 @@ function setDefaultValues() {
                     $source_form = $Proj->metadata[$source_field]['form_name'];
 
                     // Getting previous event ID.
+                    // For the first instance of a repeating event/form, the previous event is the closest saved event/form prior to the current event/form.
+                    // For the nth instance of a repeating event/form, the previous event is the current event.
                     foreach ($events as $event) {
-                        if ($event == $_GET['event_id']) {
+                        if (!in_array($source_form, $Proj->eventsForms[$event])) {
                             break;
                         }
 
-                        if (in_array($source_form, $Proj->eventsForms[$event])) {
+                        // Only break for the first instance of an event/form.
+                        // First instance events or forms need data from previous events/forms, whereas, nth repeating events/forms depend on data from the current event i.e., $_GET['event_id']
+                        if ($event == $_GET['event_id'] && $this->isFirstRepeatEventOrForm()) {
+                            break;
+                        } 
+
+                        // Set the previous event for first instance of a repeat event/form to the previous event.
+                        // Set the previous event for the nth instance of a repeat event/form to the current event.
+                        if ($this->isFirstRepeatEventOrForm()) {
                             $prev_event = $event;
-                        }
+                        } elseif ($this->isNthRepeatEventOrForm()) {
+                            $prev_event = $_GET['event_id'];
+                            break;
+                        } 
                     }
 
                     if (!$prev_event) {
                         continue;
                     }
 
                     // Getting previous event value.
-                    if (isset($data[$prev_event][$source_field])) {
-                        $default_value = $data[$prev_event][$source_field];
-                    } elseif (isset($data['repeat_instances'][$prev_event][""])) {
+                    // isset returns true for event instances when $prev_event_field_value is equal to an empty string ("").
+                    // An additional check to verify the value is not empty is required.
+                    $prev_event_field_value = $data[$prev_event][$source_field];
+
+                    // The object returned by $instances = $data['repeat_instances'][$event] changes dependening on if the data is from an event or a form.
+                    // For repeating events, the key for $instances is an empty string ("") i.e., ["": ...]
+                    // For repeating forms, the key for $instances is the form name i.e., ["baseline_data": ...]
+                    $instances = ($data['repeat_instances'][$prev_event][$source_form]) ?? ($data['repeat_instances'][$prev_event][""]);
```

Comment:
Interesting, I didn't know about the `??` operator. We'll need to bump the minimum PHP version to 7 in the `config.json`.